Code/RDGeneral/ConcurrentQueue.h

```diff
@@ -0,0 +1,102 @@
+#ifdef RDK_THREADSAFE_SSS
+#ifndef CONCURRENT_QUEUE
+#define CONCURRENT_QUEUE
+#include <condition_variable>
+#include <queue>
+#include <thread>
+
+namespace RDKit {
+template <class E>
+class ConcurrentQueue {
+ private:
+  size_t capacity;
+  bool done;
+  std::mutex lock;
+  //! we need two condition variables to establish communication
+  //! between popping (consumer threads) and pushing (producer threads)
+  std::condition_variable goPush, goPop;
+  std::queue<E> q;
+
+ public:
+  ConcurrentQueue<E>(size_t capacity) : capacity(capacity), done(false) {}
+  //! tries to push an element into the queue if it is not full without
+  //! modifying the variable element
+  void push(const E& element);
+
+  //! tries to pop an element from the queue if it is not empty and not done
+  //! the boolean value indicates the whether popping is successful
+  bool pop(E& element);
+
+  //! checks whether the ConcurrentQueue is empty
+  bool isEmpty();
+
+  //! returns the value of the variable done
+  bool getDone();
+
+  //! sets the variable done = true
+  void setDone();
+
+  //! returns the capacity of the ConcurrentQueue
+  size_t limit();
```

Comment:
this method should be const

Reply:
To be honest though, I'm not sure you need this method at all. It's kind of an implementation detail and I'm not sure why client code would need it.